### PR TITLE
Permettre le test de la non-calculabilité

### LINKED ISSUE #34: Gérer la non-calculabilité
Certaines simulations mènent à des résultats non calculables. Ces situations ne sont pas gérées par Ludwig. On ne peut donc pas tester ces cas de non-calculabilité, qui sont des résultats légitimes.
Le comportement actuel est de ne pas tester ces résultats.

- - - -

Un résultat non calculable est représenté par une chaîne de caractères identifiant de manière unique le motif de non-calculabilité.

### PR DESCRIPTION
**Pull request by MattiSG**
Fixes #34.

uncomputability

### CODE REVIEW

**Comment by fpagnoux** [comment]
Le rendu est un peu différent chez moi : 
image

File: app/scripts/services/acceptanceTest.js [hunk 1]
```diff
@@ -29,17 +29,21 @@ angular.module('ludwig').factory('AcceptanceTestsService', function($q, $http, $
         }
     };
 
-    function displayValue(value, unit) {
+    function displayValueFor(id, value, unit) {
```
Comment (fpagnoux) [inline app/scripts/services/acceptanceTest.js]: Quitte à passer `id`, on peut peut-être récupérer aussi `unit` à partir de `possibleValues` sans le passer en paramètre?

File: app/views/form.html [hunk 2]
```diff
@@ -46,14 +46,11 @@
     </keyword-input>
   </div>
 
-  <h3>Droits attendus</h3>
+  <h3>Valeurs attendues</h3>
 
-  <p>
-    Attention : veillez à supprimer ici les droits que vous ne testez pas.
-    <br>Veillez aussi à ajouter les droits dont vous testez que le montant calculé doit être nul (en mettant "0" comme valeur attendue).
```
Comment (fpagnoux) [inline app/views/form.html]: We are losing this precision. Do you think it was unnecessary ?

File: app/views/form.html [hunk 3]
```diff
@@ -46,14 +46,11 @@
...
-  </p>
+  <p>Supprimez du test les valeurs dont vous n’êtes pas certain·e.</p>
```
Comment (fpagnoux) [inline app/views/form.html]: :+1: :smile: 
It's nice to see the 'droits' disappear !

File: app/views/form.html (hunk 2 above)
Comment (MattiSG) [inline app/views/form.html]: All values are there by default, and I don't remember cases where a user would not have tested a null amount. The issue seems to be more often that they don't remove values they don't intend to test.

Unless we have evidence of user need, I'd remove as much text as possible to simplify the process.
You have interacted over Ludwig more often than I have. What is your impression?

File: app/views/form.html (hunk 3 above)
Comment (MattiSG) [inline app/views/form.html]: I did it to follow your great start in the code  :)

File: app/scripts/services/acceptanceTest.js (hunk 1 above)
Comment (MattiSG) [inline app/scripts/services/acceptanceTest.js]: :+1: Working on it.

**Comment by MattiSG** [comment]
> Le rendu est un peu différent chez moi

Ah oui, c'est un peu vilain. Je vais essayer de mettre "devrait" dans le menu déroulant, qu'en penses-tu ?

File: app/views/form.html (hunk 2 above)
Comment (fpagnoux) [inline app/views/form.html]: The only problematic case would be if :
- The user wants to make sure a value is null
- When s·he creates the test, mes-aides is right, and the value is null
- s·he doesn't explicitly set it to null
- Later there is a regression and the result becomes non-null. It is not detected.

This doesn't seem to be so likely (if user want to test a specific value, I guess they'll think about setting it to 0). So ok for removing the text.

**Comment by fpagnoux** [comment]
Why not !

**Comment by MattiSG** [comment]
capture d ecran 2015-12-16 a 16 13 33

**Comment by fpagnoux** [comment]
It's acceptable imo

**Comment by fpagnoux** [comment]
Tested, GTM

**Pull request merged**